llvm/lib/Transforms/Scalar/NewGVN.cpp

```diff
@@ -374,9 +381,10 @@ class CongruenceClass {
     if (this == Other)
       return true;
 
-    if (std::tie(StoreCount, RepLeader, RepStoredValue, RepMemoryAccess) !=
-        std::tie(Other->StoreCount, Other->RepLeader, Other->RepStoredValue,
-                 Other->RepMemoryAccess))
+    if (std::tie(StoreCount, RepLeader.first, RepStoredValue,
```

Comment:
why not compare the whole RepLeader instead of just .first?

Reply:
Done